sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -675,6 +675,16 @@ object SQLConf {
       "disable logging or -1 to apply no limit.")
     .createWithDefault(1000)
 
+  val CODEGEN_TRY_INLINE_ALL_STATES =
+    buildConf("spark.sql.codegen.tryInlineAllStates")
+    .internal()
+    .doc("When adding mutable states during code generation, whether or not we should try to " +
+      "inline all the states. If this config is false, we only try to inline primitive stats, " +
+      "so that primitive states are more likely to be inlined. Set this config to true to make " +
+      "the behavior same as Spark 2.2.")
```

Comment:
I think it only behaves the same before we hit the threshold?

Reply:
yea, let me improve it.